rpc/message.go

```diff
@@ -144,6 +188,21 @@ func (r *rpcResponseMessage) DecodeMessage(l int, d *fieldDecoder, _ *protocolHa
 	if r.c.res == nil {
 		return nil
 	}
+
+	compressor := compressorCacher.getCompressor(r.c.ctype)
+	if compressor != nil {
+		var compressed []byte
```

Comment:
Shouldn't this be a `bytes.Buffer`, so the length doesn't matter?

Reply:
```
Error in transport: RPC error. type: Response, method: test.1.testp.GetNConstants, length: 4, error: error decoding message field at position 3, error: msgpack decode error [pos 4]: only encoded map or array can be decoded into a struct
```

Reply:
Oh right, I guess it reallocates the `[]byte`.